src/Storage.js

```diff
@@ -0,0 +1,42 @@
+import redis from 'redis'
+
+class Storage {
+
+  constructor(prefix, redisOpts){
+    const opts = redisOpts || {}
+    if (!redis.createClient) throw new Error('Storage: Redis is not available in this enviroment, use functionality which does not require persistent storage')
+    this.client = redis.createClient(opts)
+    this.client.on("error", err => {console.log("Error " + err)})
+    this.prefix = prefix ? `${prefix}:` : ''
+    this.key = pairId => (`${this.prefix}${pairId}`)
+  }
+
+  get(key) {
+    return new Promise((resolve, reject) => {
+      this.client.get(this.key(key), (err, res) => {
+        if (err) reject(err)
+        return resolve(res)
+      })
+    })
+  }
+
+  set(key, value) {
+    return new Promise((resolve, reject) => {
+      this.client.set(this.key(key), value, (err, res) => {
+        if (err) reject(err)
+        return resolve(res)
+      })
+    })
+  }
+
+  del(key) {
```

Comment:
yeah its possible, but I added del such that it the storage interface is more clear for anyone looking to implement their own. I also don't think the high level code would expect it to also delete when you call get